### PR TITLE
pythonPackages.construct: Remove pytest-benchmark check dependency

### PR DESCRIPTION
**Pull request by zhaofengli**
###### Motivation for this change

No need to pull it in at all when the benchmarks are skipped anyways.

<del>Also disable one of its dependencies, py-cpuinfo, on RISC-V since it's unsupported.</del> Dropping this for now since upstream has finally added RISC-V support recently, pending release.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126173` at 58f35a6a run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages marked as broken and skipped:</summary>
  <ul>
    <li>cassandra_2_1</li>
    <li>cassandra_2_2</li>
    <li>hadoop</li>
    <li>hadoop_2_7</li>
    <li>hadoop_2_8</li>
    <li>hadoop_2_9</li>
    <li>impressive</li>
    <li>linuxPackages_hardkernel_4_14.virtualbox</li>
    <li>linuxPackages_hardkernel_latest.virtualbox</li>
    <li>odpdown</li>
    <li>opentsdb</li>
    <li>pig</li>
    <li>spark</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/8fc4e13b7d33d0ef588434d4c5bf0a67">python39Packages.brotlipy</a></li>
    <li><a href="https://gist.github.com/ed4f9e488670a04c887e0da40625bff7">python39Packages.construct</a></li>
  </ul>
</details>
<details>
  <summary>1175 packages skipped due to time constraints:</summary>
  <ul>
    <li>DisnixWebService</li>
    <li>R</li>
    <li>abcl</li>
    <li>adapta-gtk-theme</li>
    <li>adoptopenjdk-icedtea-web</li>
    <li>aiodnsbrute</li>
    <li>alloy (alloy5)</li>
    <li>alloy4</li>
    <li>amidst</li>
    <li>ansible (ansible_2_11 ,python38Packages.ansible-core)</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>6 packages built successfully:</summary>
  <ul>
    <li>python38Packages.brotlipy</li>
    <li>python38Packages.construct</li>
    <li>python38Packages.httpbin</li>
    <li>python38Packages.pytest-httpbin</li>
    <li>python38Packages.pytest-vcr</li>
    <li>python38Packages.vcrpy</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `six`.
Near pkgs/development/python-modules/construct/default.nix:2:3:
```
  |
2 | , six, pytestCheckHook, pytest-benchmark, numpy, arrow, ruamel_yaml
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `pytest-benchmark`.
Near pkgs/development/python-modules/construct/default.nix:2:25:
```
  |
2 | , six, pytestCheckHook, pytest-benchmark, numpy, arrow, ruamel_yaml
  |                         ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/python-modules/construct/2.10.54.nix [hunk 1]
```diff
@@ -14,7 +14,7 @@ buildPythonPackage rec {
     sha256 = "1mqspsn6bf3ibvih1zna2glkg8iw7vy5zg9gzg0d1m8zcndk2c48";
   };
 
-  checkInputs = [ pytestCheckHook pytest-benchmark enum34 numpy ];
+  checkInputs = [ pytestCheckHook enum34 numpy ];
```
Comment (fabaff) [inline pkgs/development/python-modules/construct/2.10.54.nix:17]: `enum34` can be removed as well. The module only supports [Python > 3.6](https://github.com/construct/construct/blob/master/setup.py#L19).

Could you please add `disabled = pythonOlder "3.6";` to make sure that we respect upstream's limit of the Python release?

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/python-modules/construct/2.10.54.nix (hunk 1 above)
Comment (zhaofengli) [reply pkgs/development/python-modules/construct/2.10.54.nix:17]: Actually, I just checked and 2.10.54 is only there for Python 2 support (search for `2.10.54` in `all-packages.nix`).

Comment (fabaff) [reply pkgs/development/python-modules/construct/2.10.54.nix:17]: You are right.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 126173 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>python3Packages.construct</li>  </ul></details>

**Pull request merged**